src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerAssemblyLoader.Core.cs

```diff
@@ -18,58 +18,71 @@
 
 namespace Microsoft.CodeAnalysis
 {
-    internal enum AnalyzerLoadOption
+    internal sealed partial class AnalyzerAssemblyLoader
     {
-        /// <summary>
-        /// Once the assembly path is chosen, load it directly from disk at that location
-        /// </summary>
-        LoadFromDisk,
+        internal static IAnalyzerAssemblyResolver DiskAnalyzerAssemblyResolver => DiskResolver.Instance;
+        internal static IAnalyzerAssemblyResolver StreamAnalyzerAssemblyResolver => StreamResolver.Instance;
 
         /// <summary>
-        /// Once the assembly path is chosen, read the contents of disk and load from memory
+        /// Map of directories to load contexts.
```

Comment:
May be worth commenting which directory this is. (Pre-mapping, post mapping, etc.)